src/app/rpmostree-builtin-status.c

```diff
@@ -498,25 +503,54 @@ status_generic (RPMOSTreeSysroot *sysroot_proxy,
               g_variant_get_child (gv_nevra_new, 1, "&s", &name_new);
               g_variant_get_child (gv_nevra_old, 1, "&s", &name_old);
 
-              if (str->len)
-                g_string_append (str, ", ");
-
               /* if pkgnames match, print a nicer version like treediff */
               if (g_str_equal (name_new, name_old))
                 {
-                  g_autofree char *old_evr = gv_nevra_to_evr (gv_nevra_old);
-                  g_autofree char *new_evr = gv_nevra_to_evr (gv_nevra_new);
-                  g_string_append_printf (str, "%s %s -> %s", name_new, old_evr, new_evr);
+                  /* let's just use str as a scratchpad to avoid excessive mallocs; the str
+                   * needs to be stretched anyway for the final output */
+                  gsize original_size = str->len;
+                  gv_nevra_to_evr (str, gv_nevra_old);
+                  g_string_append (str, " -> ");
+                  gv_nevra_to_evr (str, gv_nevra_new);
+                  const char *diff = str->str + original_size;
+                  GPtrArray *pkgs = g_hash_table_lookup (grouped_diffs, diff);
+                  if (!pkgs)
+                    {
+                      pkgs = g_ptr_array_new_with_free_func (g_free);
+                      g_hash_table_insert (grouped_diffs, g_strdup (diff), pkgs);
+                    }
+                  g_ptr_array_add (pkgs, g_strdup (name_new));
+                  g_string_truncate (str, original_size);
                 }
               else
                 {
+                  if (str->len)
+                    g_string_append (str, ", ");
+
                   const char *nevra_old;
                   g_variant_get_child (gv_nevra_old, 0, "&s", &nevra_old);
                   g_string_append_printf (str, "%s -> %s", nevra_old, nevra_new);
                 }
               g_ptr_array_add (active_replacements, g_strdup (nevra_new));
             }
+
+          GLNX_HASH_TABLE_FOREACH_KV (grouped_diffs, const char*, diff, GPtrArray*, pkgs)
+            {
+              if (str->len)
+                g_string_append (str, ", ");
+              for (guint i = 0, n = pkgs->len; i < n; i++)
+                {
+                  const char *pkgname = g_ptr_array_index (pkgs, i);
+                  if (i > 0)
+                    g_string_append_c (str, ' ');
+                  g_string_append (str, pkgname);
+                }
+              g_string_append_c (str, ' ');
+              g_string_append (str, diff);
+            }
+
           g_ptr_array_add (active_replacements, NULL);
+
```

Comment:
Hey, not too sure how the style generally works, so do we need to add this extra line here?

Reply:
Mostly personal preference. It separates bits that aren't directly related to each other. I can remove it if you feel strongly about it. :)

Reply:
ah, dw about it :). I was just trying to make sure I am still caught up with the style of the codebase :p.